src/storage/MonitoringStore.ts

```diff
@@ -0,0 +1,49 @@
+import { EventEmitter } from 'events';
+import type { Patch } from '../ldp/http/Patch';
+import type { Representation } from '../ldp/representation/Representation';
+import type { RepresentationPreferences } from '../ldp/representation/RepresentationPreferences';
+import type { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import type { Conditions } from './Conditions';
+import type { ResourceStore } from './ResourceStore';
+
+/**
+ * Store that notifies listeners of changes to its source
+ * by emitting a `changed` event.
+ */
+export class MonitoringStore<T extends ResourceStore = ResourceStore>
+  extends EventEmitter implements ResourceStore {
+  private readonly source: T;
```

Comment:
What is the point of having a generic here instead of just having `source: ResourceStore` since it doesn't actually expose anything related to the type?

Reply:
A copy/paste from the passthrough store; for case of future extension I suppose.